### PR TITLE
Add status extensions

### PR DESCRIPTION
**Pull request by yashpola**
- #93 and #94 
- GUI change: `accepted` and `rejected` statuses for applicants have green and red backing respectively

### CODE REVIEW

**Review by headcube1 — APPROVED** [review]
LGTM

File: src/main/java/seedu/address/model/person/Applicant.java [hunk 1]
```diff
@@ -12,16 +14,16 @@
 public class Applicant extends Person {
 
     private final Type type = Type.APPLICANT;
-
-    private ApplicantStatus status;
+    private ApplicantStatus previousStatus;
+    private ApplicantStatus currentStatus;
```
Comment (macareonie) [inline src/main/java/seedu/address/model/person/Applicant.java:18]: nice splitting status into previous and current for more clarity

File: src/main/java/seedu/address/model/person/Person.java [hunk 2]
```diff
@@ -72,7 +73,15 @@ public String getPersonType() {
...
+
+    public void updateCurrentStatusToReflectInterview(Model model, Person applicantScheduled) {
+    }
+
+    public void revertCurrentStatus(Model model) {
+    }
+
+    public String getCurrentStatus() {
         return "";
     }
```
Comment (macareonie) [inline src/main/java/seedu/address/model/person/Person.java:86]: With the addition of more of such methods, is it ever feasible changing this class to abstract instead and have the respective subclasses override them?

**Review by macareonie — APPROVED** [review]
👍

**Review by headcube1 — APPROVED** [review]

**Pull request merged**